frontend/src/service/ChangeLog.service.ts

```diff
@@ -1,73 +1,72 @@
-import { Update } from '../entity/Update';
-
-type Time = number;
+import { ChangeLog } from '../entity/ChangeLog';
+import { ChangeLogGraphQLApi } from './ChangeLogGraphQL.api';
+import { ErrorService, Err } from './Error.service';
+import { Change } from '../entity/Change';
 
 export class ChangeLogService {
+  constructor(
+    private changeLogGraphQLApi: ChangeLogGraphQLApi,
+    private errorService: ErrorService
+  ) {}
+
   hasUpdates(): Promise<boolean> {
     return new Promise(async (resolve, reject) => {
-      const changeLog = await this.getChangeLog();
-      const lastSeenAt = await this.getLastSeenChangeLog();
-      if (!changeLog) {
-        resolve(false);
-        return;
+      let changeLog;
+      try {
+        changeLog = await this.changeLogGraphQLApi.getChangeLog();
```

Comment:
I though if you move the change attribute validation into `getChangeLog` and call `this.getChangeLog()` here, then you don't have to sort again in this function. Please let me know if I am wrong.

Reply:
Discussed offline. Keep it as it is.